### ISSUE TITLE
tests failing with zstd 0.8.2+zstd.1.5.0

### ISSUE THREAD

**Issue by ahornby** [Nemo157/async-compression#122]
I've tried updating async-compression to zstd 1.5.0 in https://github.com/ahornby/async-compression/tree/zstd_1.5.x, but I see test failures when it picks up zstd 0.8.2.  If I force it 0.8.1 it works

Reproduces with
~/local/async-compression$ cargo test --features zstd,tokio zstd

I don't know if this is an async-compression or zstd-rs issue.

**Comment by Nemo157** [comment]
That is concerning, those are patch updates so can happen for any downstream users. Taking a look into where the issue is now.

**Comment by Nemo157** [comment]
See also https://github.com/gyscos/zstd-rs/issues/122, it looks like this is probably an upstream `zstd` issue.

**Comment by Nemo157** [comment]
This is fixed in `zstd` 0.8.3:

```
> cargo update -p zstd -p zstd-safe
    Updating crates.io index
    Updating zstd v0.8.0+zstd.1.4.9 -> v0.8.3+zstd.1.5.0
    Updating zstd-safe v4.0.0+zstd.1.4.9 -> v4.1.0+zstd.1.5.0
    Updating zstd-sys v1.5.0+zstd.1.4.9 -> v1.6.0+zstd.1.5.0

> cargo test --features zstd,tokio --test zstd
[...]
test result: ok. 34 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.15s
```

Thanks for opening the issue, I'm going to try to automate checking for regressions like this to make sure they get noticed straight away :grin:
